### ISSUE TITLE
Can't bind to 'formControl' since it isn't a known property of 'ngx-select'.

### ISSUE THREAD

**Issue by abelardolg** [optimistex/ngx-select-ex#51]
Hi,
When I used the single select, I got this error.

> HTML
* It's the same contained in the doc.

> TypeScript:
* I modified it slightly since I passed it a list containing objects with these fields: id, ds and selected. 
* The local variable '_ngxDefault' stores 'this.lista[idx].ds' ('lista' is my collection of items).

> My feature module:
* import {INgxSelectOptions, NgxSelectModule} from 'ngx-select-ex';
* ` const CustomSelectOptions: INgxSelectOptions = { // Check the interface fo more options
  optionValueField: 'id',
  optionTextField: 'ds'
};
`
* NgxSelectModule.forRoot(CustomSelectOptions)

Best regards.

**Comment by optimistex** [comment]
@abelardolg  My feature module...
It seems that you using the `ngx-select-ex` in few different modules. I do not sure that the `NgxSelectModule.forRoot()` will be work correctly from submodule. Please, check an angular documentation.

**Comment by optimistex** [comment]
I can't install it.
![image](https://user-images.githubusercontent.com/11228640/38255347-48e38ffa-3764-11e8-8e0b-c56ab3399b48.png)

**Comment by abelardolg** [comment]
Sorry, I hope this .zip works!:
[Uploading app-mantenimiento-tablas.zip…]()

**Comment by abelardolg** [comment]
The component is in this path:
\src\app\templates\components\select-simple-sge
\src\app\test

**Comment by optimistex** [comment]
Looks like a bug.
I'll fix it.

**Comment by abelardolg** [comment]
Sure. :) 

I will wait the future versions and I will check it in order to use with: 

- my list full of {id: string, ds: string, selected: boolean} objects and to search for by 'id' and by 'ds'.

- the down key (I tested it but I had to press several times to show the items).

I will back within a few days. Brs.

**Comment by optimistex** [comment]
It's done!

Solution:
1. Install the new release
2. Add the attribute `[optionValueField]="id"` for comfortable output
3. The comparison `search === item.data.id` change to `+search === item.data.id` for cast a string to a number.

Good luck! :+1:

**Comment by abelardolg** [comment]
Hi again @optimistex !
Thanks for your support. 👍 

Now, I am trying to know why my list of items -> {id: number, ds: string, region: string) isn't loaded inside the select. 
According to my code (I transferred it to you yesterday), do you know why the select doesn't load the list? 🤔 
This fragment works:
```
this._ngxDefault = this.listaSGE[idx].ds;
        console.log(this._ngxDefault);
```
but the select doesn't load the collection of items. 
Even when I create an array of strings I am not be able to load it into the select. 🤔 
![image](https://user-images.githubusercontent.com/22522883/38293406-a7fcbeca-37e7-11e8-9fd5-e1bc18599e16.png)

**Comment by optimistex** [comment]
Look at your `searchCallBack`. It is almost always returning the `false` that hides all items.

```js
public searchCallback = (search: string, item: INgxSelectOption) => ((search === item.data.id) || (item.data.ds === search));
```

`search === item.data.id` - always will be `FALSE`. Because the `search` is a `string` and the `item.data.id` is a `number` and you have a strict compare `===`. Just add the `+` before the `search` for converting it into a number!

Then use the code:
```js
public searchCallback = (search: string, item: INgxSelectOption) => ((+search === item.data.id) || (item.data.ds === search));
```

**Comment by abelardolg** [comment]
Hi again,
I don't think it is the cause because I tested it as is you have written it but it still doesn't work.

I am still investigating the cause. 

Brs.

**Comment by abelardolg** [comment]
[optionValueField]="id"
->
ERROR in select-simple-sge\select-simple-sge.component.html(7,17): : Property 'id' does not exist on type 'SelectSimpleSgeComponent'.

**Comment by optimistex** [comment]
Upload your project to the GitHub or some similar GIT system and give me the access.
I'll try to fix your issue.

I do not want to fix it in the zip archive because we cant to synchronize it correctly.

**Comment by abelardolg** [comment]
I will upload to my webpage tonight. :)

**Comment by abelardolg** [comment]
Hi again,

I uploaded a fresh version of the project, from scratch.

You can check it [here](http://www.abelardolg.com/lab/angular/)

I have pressed the Arrow Down key but the list didn't open. 

I am still investigating how to search for by an id and by a text.

Brs.

**Comment by optimistex** [comment]
I see the page, but I don't see the code. Compiled code is not better for human reading...

**Comment by abelardolg** [comment]
You can test it again. It works: you can type '1' and it should appear 'uno' and viceversa (type 'uno' and you should see '1'...but Arrow Down isn't still working. 
Delete cookies...clean cache...

**Comment by optimistex** [comment]
For working the ArrowDown you have to change the options of the module. Did you do it?

**Comment by abelardolg** [comment]
Yes:
const CustomSelectOptions: INgxSelectOptions = { // Check the interface for more options
  optionValueField: 'id',
  optionTextField: 'ds',
  keyCodeToOptionsOpen: 'ArrowDown'

**Comment by optimistex** [comment]
Show me full code of it.

**Comment by abelardolg** [comment]
I will upload it.

**Comment by optimistex** [comment]
@abelardolg I will wait ))

**Comment by abelardolg** [comment]
I think it can be a cache issue.

**Comment by optimistex** [comment]
@abelardolg I think no it is not

**Comment by optimistex** [comment]
Again:
![image](https://user-images.githubusercontent.com/11228640/38332914-c794a892-385f-11e8-99bb-941c6e241307.png)

**Comment by abelardolg** [comment]
[helloworld-select.zip](https://github.com/optimistex/ngx-select-ex/files/1877517/helloworld-select.zip)

**Comment by optimistex** [comment]
Really?
![image](https://user-images.githubusercontent.com/11228640/38333051-1fa2c1c2-3860-11e8-8691-26c8126ce9ea.png)

**Comment by optimistex** [comment]
Okay. It is installed...

**Comment by optimistex** [comment]
I do not see a trouble.
I did:
* npm i
* npm start
* open the url http://localhost:4200
* click to select (it opens)
* press Esc (to close but do not lost the focus)
* press ArrowDown (it opens again)

All Okay. 
For understanding. The events works from focused control. If it is not focused then it will not work. And it is not an issue of the component. It is an issue of understanding "How works focus events in an application - does not matter what an application!"

**Comment by abelardolg** [comment]
I am testing it with Opera and Google Chrome for High Sierra.
When I click on the down arrow (located at the right side of the select) I don't see the list of items.
In fact, when I press the Arrow Down key I don't see the output in the console log. I don't know if there are differences Windows and Mac related to key events. 
Tomorrow will be another day. Thank so much for your time. PRICELESS.

**Comment by optimistex** [comment]
> if there are differences Windows
I work on Ubuntu (Linux). It is a friendly system for MacOS

**Comment by optimistex** [comment]
Just pressing keyEsc/keyDown. No more actions:
![image](https://user-images.githubusercontent.com/11228640/38334524-3ec0553e-3864-11e8-8ee3-22c1525f8691.png)

**Comment by optimistex** [comment]
@abelardolg Look! I think We have a big issue.
**KeyboardEvent.keyCode** 
It is deprecated and soon will not work:
> Deprecated
This feature has been removed from the Web standards. Though some browsers may still support it, it is in the process of being dropped. Avoid using it and update existing code if possible; see the compatibility table at the bottom of this page to guide your decision. Be aware that this feature may cease to work at any time.

But not it works everywhere. See `Browser compatibility`: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode
____________
KeyboardEvent.code
Now I switched all events to this code.
But, it is not compatible with a few browsers:
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code

=================
I have no any idea what to do. And I very want to dirty swear!

Do you have some idea? )))

**Comment by optimistex** [comment]
But. It must work in the Chrome browser!

**Comment by abelardolg** [comment]
Hi there,

On the one hand:
I have tested in the Google browser under Windows what I uploaded yesterday. 
My experience: 
- when I pressed the Arrow down (previously I got the focus of the component), the "No results found" message was showed (I didn't see the items).
- When I typed "o", none of the items which contains this letter ("uno", "dos", "cuatro", "cinco", "ocho") was showed; I had to type the entire word "one" to select it. I am going to investigate how to do this.

Finally:
The best approach to solve the big issue of the key events is to build a wrapper to detect which browser, which version and which OS is being used by the user in order to bring them the best experience. There is a page (at the moment I don't remember it) which provides compatibility information related to a specific feature. 

I will return to this discussion with solutions (hope) 😄 

Brs.
